File: LeetCode/Easy/Next Greater Element I/Java/nextGreaterElement.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> index = new HashMap<>();
        for (int i=0; i<nums2.length; i++) index.put(nums2[i], i);
        int[] res = new int[nums1.length];
        for (int i=0; i<nums1.length; i++) {
            int ind = index.get(nums1[i]);
            int val = -1;
            for (int j=ind; j<nums2.length; j++) {
                if (nums1[i] < nums2[j]) {
                    val = nums2[j];
                    break;
                }
            }
            res[i] = val;
        }
        return res;
    }
}

File: LeetCode/Easy/Next Greater Element I/c++/next_greater_element.cc
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> u_map;
        vector<int> res(nums1.size());

        for (int i = 0; i < nums2.size(); i++) {
            u_map[nums2[i]] = i;
        }

        for (int i = 0; i < nums1.size(); i++) {
            int ind = u_map[nums1[i]];
            int num = nums1[i];
            for (int j = ind + 1; j < nums2.size(); j++) {
                if (num < nums2[j]) {
                    num = nums2[j];
                    break;
                }
            }

            res[i] = (num == nums1[i] ? -1 : num);
        }
        
        return res;
    }
};
